packages/block-library/src/search/edit.native.js

```diff
@@ -155,21 +157,35 @@ export default function SearchEdit( {
 	);
 
 	const mergeWithBorderStyle = ( style ) => {
-		return { ...style, ...styles.border };
+		const baseStyle = getStylesFromColorScheme(
+			styles.border,
+			styles.borderDark
+		);
+		return { ...style, ...baseStyle };
```

Comment:
What do you think if we just have the base border styles in a const so we can use a simpler approach to merge the styles? If we replace the whole `mergeWithBorderStyle` function with this:

```suggestion
	const isButtonInside = buttonPosition === 'button-inside';
	const borderStyle = getStylesFromColorScheme(
		styles.border,
		styles.borderDark
	);
```

I added an `isButtonInside` const in the above suggestion since it's used twice.

Reply:
Sorry, read and implemented the first suggestion immediately before I saw the code suggestions you made so they became invalidated. This was also implemented in 1b17b727a80a29459f4c66790c531c341c7b6825